posthog/src/test/java/com/posthog/PostHogStatelessTest.kt

```diff
@@ -796,6 +796,61 @@ internal class PostHogStatelessTest {
         assertEquals(0, mockQueue.events.size)
     }
 
+    @Test
+    fun `feature flag called events are deduplicated per distinct ID`() {
+        val mockQueue = MockQueue()
+        val mockFeatureFlags = MockFeatureFlags()
+        mockFeatureFlags.setFlag("test_flag", "variant_a")
+
+        sut = createStatelessInstance()
+        config = createConfig(sendFeatureFlagEvent = true)
+
+        sut.setup(config)
+        sut.setMockQueue(mockQueue)
+
+        sut.setMockFeatureFlags(mockFeatureFlags)
+
+        // Same user accessing same flag multiple times - should deduplicate
+        sut.getFeatureFlagStateless("user123", "test_flag")
+        sut.getFeatureFlagStateless("user123", "test_flag")
+
+        // Different user accessing same flag - should send separate event
+        sut.getFeatureFlagStateless("user456", "test_flag")
+
+        // Should generate two $feature_flag_called events (one per distinct ID)
+        assertEquals(2, mockQueue.events.size)
+        assertEquals("user123", mockQueue.events[0].distinctId)
+        assertEquals("variant_a", mockQueue.events[0].properties!!["${'$'}feature_flag_response"])
+        assertEquals("user456", mockQueue.events[1].distinctId)
+        assertEquals("variant_a", mockQueue.events[1].properties!!["${'$'}feature_flag_response"])
+    }
+
+    @Test
+    fun `feature flag called events are sent for different values of same flag`() {
```

Comment:
Good test!

This covers when variants are different, but for completeness, I'd also make sure that two events are sent for same `distinct_id` and same flag when the evaluated value changes (aka `true`, `false`, and `null`).